### ISSUE TITLE
Run-Time Check Failure #2 - Stack around the variable 'node' was corrupted.

### ISSUE THREAD

**Issue by jeffcutsinger** [dictation-toolbox/natlink#52]
I recently upgraded my natlink and its dependencies, and now whenever I issue a command with caster, I get this messagebox:

```
---------------------------
Microsoft Visual C++ Runtime Library
---------------------------
Debug Error!

Program: C:\Python38-32\Lib\site-packages\natlinkcore\natlink.pyd
Module: C:\Python38-32\Lib\site-packages\natlinkcore\natlink.pyd
File: 

Run-Time Check Failure #2 - Stack around the variable 'node' was corrupted.

(Press Retry to debug the application)

---------------------------
Abort   Retry   Ignore   
---------------------------

```

I tried downgrading natlink (but not its dependencies), and the behavior continues. I tried 0.0.0.1.9-0.0.0.1.2, with the exception of 0.0.0.1.3.

As to the versions of natlink's dependencies:

```
C:\WINDOWS\system32>pip install --no-cache --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple natlink
Looking in indexes: https://test.pypi.org/simple/, https://pypi.org/simple
Requirement already satisfied: natlink in c:\python38-32\lib\site-packages (0.0.0.1.9)
Requirement already satisfied: wxpython==4.0.7 in c:\python38-32\lib\site-packages (from natlink) (4.0.7)
Requirement already satisfied: pywin32>=300 in c:\python38-32\lib\site-packages (from natlink) (301)
Requirement already satisfied: debugpy in c:\python38-32\lib\site-packages (from natlink) (1.3.0)
Requirement already satisfied: pillow in c:\python38-32\lib\site-packages (from wxpython==4.0.7->natlink) (8.3.1)
Requirement already satisfied: numpy in c:\python38-32\lib\site-packages (from wxpython==4.0.7->natlink) (1.21.1)
Requirement already satisfied: six in c:\python38-32\lib\site-packages (from wxpython==4.0.7->natlink) (1.16.0)
```

My setup had been working beforehand, with the exception of some occasional glitches which I'd hoped an upgrade would alleviate. I'm using Dragon Professional Individual 15.

**Comment by jeffcutsinger** [comment]
Remembered I had a working installation on another machine. I pulled its list of versions and replicated them, and now it's working. Here's what I ended up with:

```
Requirement already satisfied: natlink in c:\python38-32\lib\site-packages (0.0.0.1.8)
Requirement already satisfied: vocola2 in c:\python38-32\lib\site-packages (0.1.0.5)
Requirement already satisfied: unimacro in c:\python38-32\lib\site-packages (0.0.3.8)
Requirement already satisfied: pywin32>=300 in c:\python38-32\lib\site-packages (from natlink) (301)
Requirement already satisfied: debugpy in c:\python38-32\lib\site-packages (from natlink) (1.3.0)
Requirement already satisfied: wxpython==4.0.7 in c:\python38-32\lib\site-packages (from natlink) (4.0.7)
Requirement already satisfied: six in c:\python38-32\lib\site-packages (from wxpython==4.0.7->natlink) (1.16.0)
Requirement already satisfied: pillow in c:\python38-32\lib\site-packages (from wxpython==4.0.7->natlink) (8.2.0)
Requirement already satisfied: numpy in c:\python38-32\lib\site-packages (from wxpython==4.0.7->natlink) (1.20.3)
```

Feel free to close this if you want, otherwise I'll keep an eye on this if there's something you want me to investigate.

**Comment by dougransom** [comment]
You might have experienced this if you were working from the head of the makeflitinstall branch - I put a bad natlink.pyd in there around this date, before I realized it was bad.  While I think I fixed it, the current one at the head of makeflitinstall has only been briefly tested on my machine.  Working with Quintijn to see if it is good.

For the time being install the one from testpypi.

*        pip install --no-cache --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple natlink unimacro vocola2




From: Jeff Cutsinger ***@***.***>
Sent: July 20, 2021 5:02 AM
To: dictation-toolbox/natlink ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [dictation-toolbox/natlink] Run-Time Check Failure #2 - Stack around the variable 'node' was corrupted. (#52)


I recently upgraded my natlink and its dependencies, and now whenever I issue a command with caster, I get this messagebox:

---------------------------

Microsoft Visual C++ Runtime Library

---------------------------

Debug Error!



Program: C:\Python38-32\Lib\site-packages\natlinkcore\natlink.pyd

Module: C:\Python38-32\Lib\site-packages\natlinkcore\natlink.pyd

File:



Run-Time Check Failure #2 - Stack around the variable 'node' was corrupted.



(Press Retry to debug the application)



---------------------------

Abort   Retry   Ignore

---------------------------



I tried downgrading natlink (but not its dependencies), and the behavior continues. I tried 0.0.0.1.9-0.0.0.1.2, with the exception of 0.0.0.1.3.

As to the versions of natlink's dependencies:

C:\WINDOWS\system32>pip install --no-cache --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple natlink

Looking in indexes: https://test.pypi.org/simple/, https://pypi.org/simple

Requirement already satisfied: natlink in c:\python38-32\lib\site-packages (0.0.0.1.9)

Requirement already satisfied: wxpython==4.0.7 in c:\python38-32\lib\site-packages (from natlink) (4.0.7)

Requirement already satisfied: pywin32>=300 in c:\python38-32\lib\site-packages (from natlink) (301)

Requirement already satisfied: debugpy in c:\python38-32\lib\site-packages (from natlink) (1.3.0)

Requirement already satisfied: pillow in c:\python38-32\lib\site-packages (from wxpython==4.0.7->natlink) (8.3.1)

Requirement already satisfied: numpy in c:\python38-32\lib\site-packages (from wxpython==4.0.7->natlink) (1.21.1)

Requirement already satisfied: six in c:\python38-32\lib\site-packages (from wxpython==4.0.7->natlink) (1.16.0)

My setup had been working beforehand, with the exception of some occasional glitches which I'd hoped an upgrade would alleviate. I'm using Dragon Professional Individual 15.

-

**Comment by jeffcutsinger** [comment]
All of this was done through pip and testpypi. I've never used flit.

**Comment by dougransom** [comment]
Jeff,  i think you have something working?  The fix for this i think is going to be the next natlinkcore to install via PIP.  It may be a weeks before that happens.

**Comment by quintijn** [comment]
Jeff, maybe easiest to do a pip install natlink (after a pip uninstall natlink). You should get version 5.0.0 of natlink.

Configure via natlinkconfig_gui.

Run these in cmd shell. The program natlinkconfig_gui.exe is in (\python38-32)\Scripts directory, but should run from everywhere.

We will work on install/configure instructions ASAP

**Comment by dougransom** [comment]
fixed in Natlink version: 5.1.0
